client/post-editor/editor-image-selector/preview.jsx

```diff
@@ -56,32 +56,29 @@ export class EditorImageSelectorPreview extends Component {
 		// We may not necessarily need to trigger a network request if we
 		// already have the data for the media item, so first update the state
 		this.updateImageState( () => {
-			if ( this.state.images ) {
+			const { itemIds, siteId } = this.props;
+			if ( isEqual( this.state.images.map( image => image.ID ), itemIds ) ) {
```

Comment:
It's possible in this callback that if the images were reordered, this would return falsey, but I've added another check to determine whether or not the item is already in the MediaStore so we don't fetch again if so.

https://github.com/Automattic/wp-calypso/blob/2af049b2796faea2d23fd6439ba85dadd7e8eb82/client/components/image-selector/preview.jsx#L67-L70

We could sort the arrays and compare equality to return early here, but we would still want to check the store for these files anyway to avoid duplicate fetches since the media library and dropzone handle some fetches for us.